src/frontends/tensorflow/src/op/concat.cpp

```diff
@@ -34,17 +34,47 @@ OutputVector translate_concat_op(const NodeContext& node) {
     get_const_input(node, axis_idx, &tf_concat_axis_vec);
     int64_t concat_axis = tf_concat_axis_vec[0];
 
+    // Excluding zero-dim inputs
     OutputVector ng_args;
     for (int i = concat_idx_start; i < concat_idx_stop; i++) {
         Output<Node> ng_arg = node.get_input(i);
-        ng_args.push_back(ng_arg);
-    }
+        bool valid_input = true;
 
-    auto res = make_shared<Concat>(ng_args, size_t(concat_axis));
-    set_node_name(node.get_name(), res);
-    return res->outputs();
+        if (ng_arg.get_partial_shape().is_static()) {
+          auto inp_shape = ng_arg.get_shape();
+          for (auto dim : inp_shape) {
+            if (dim == 0) {
+              valid_input = false;
+              break;
+            }
+          }
+        }
+        if (valid_input) {
+          ng_args.push_back(ng_arg);
+        }
+    }
+    // Create a Const op if all inputs are zero-dim
+    if (ng_args.empty()) {
+      int concat_axis_out_dim_value = 0;
+      ov::Output<ov::Node> ng_arg;
+      ov::Shape inp_shape;
+      for (int i = concat_idx_start; i < concat_idx_stop; i++) {
+        Output<Node> ng_arg = node.get_input(i);
+        inp_shape = ng_arg.get_shape();
+        concat_axis_out_dim_value += inp_shape[concat_axis];
+      }
+      inp_shape[concat_axis] = concat_axis_out_dim_value;
+      auto ng_et = node.get_attribute<ov::element::Type>("T");
+      auto res = make_shared<Constant>(ng_et, inp_shape, 0);
+      set_node_name(node.get_name(), res);
+      return res->outputs();
+    } else {
+      auto res = make_shared<Concat>(ng_args, size_t(concat_axis));
+      set_node_name(node.get_name(), res);
+      return res->outputs();
```

Comment:
Could you please explain what will we do in case of all input tensors to be empty? What is "T" attribute?
Actualy, you don't need to care of removing empty tensors because we have offline transformations that are removing them.

Reply:
If all input tensors are empty, it will convert the op to an empty Const op with a type which is read from the "T" attribute of the original node. In what stage these offline transformations remove the empty tensors? We were having issues with the empty tensors for some reasons I tried to explain below.

1. We convert the graph using TF FE.
2. Some parameters and some results have zero dimension. OV execution crashes if we try to execute the graph this way. Some ops do not expect empty tensors.
3. So, we remove these zero-dim parameters and results. Then we re-create the ov::Model using its constructor (passing the non-empty parameters and results only).
4. This solution works for most cases. But some other cases, ov::Model still include these zero-dim parameters since some result nodes are still connected to zero-dim parameters.
5. This is the reason why remove these empty tensors in Concat op translation.

So, is there any way we can fix this issue with offline transformations?

Reply:
@cavusmustafa, wow we have transformation RemoveConcatZero dim configured by default for all plugins: openvino\src\common\transformations\src\transformations\common_optimizations\remove_concat_zero_dim_input.cpp